tests/test_dle.py

```diff
@@ -289,15 +286,19 @@ def test_cache_blocking_imperfect_nest(blockinner):
     op1 = Operator(eqns, opt=('advanced', {'blockinner': blockinner}))
 
     # First, check the generated code
-    trees = retrieve_iteration_tree(op1._func_table['bf0'].root)
+    trees = [i for i in retrieve_iteration_tree(op1)]
```

Comment:
useless list comprehension. Drop all of them . As in, `trees = retrieve_iteration_tree(op1)` suffices